lib/iceberg/events/self_submit_orders.js

```diff
@@ -15,12 +15,18 @@ const generateOrders = require('../util/generate_orders')
  */
 const onSelfSubmitOrders = async (instance = {}) => {
   const { state = {}, h = {} } = instance
-  const { emit } = h
+  const { emit, timeout } = h
   const { args = {}, gid } = state
   const { submitDelay } = args
-  const orders = generateOrders(state)
 
-  return emit('exec:order:submit:all', gid, orders, submitDelay)
+  if (submitDelay) {
+    const [, t] = timeout(submitDelay)
```

Comment:
timeout needs to get stopped when stopping the order?

Reply:
thanks!! will add this